spark/src/test/scala/org/apache/comet/CometCastSuite.scala

```diff
@@ -738,6 +738,17 @@ class CometCastSuite extends CometTestBase with AdaptiveSparkPlanHelper {
     castTest(generateTimestamps(), DataTypes.DateType)
   }
 
+  // todo: Refactor this
+  test("xxhash64") {
+    val input = generateStrings(timestampPattern, 8).toDF("a")
+    withTempPath { dir =>
+      val data = roundtripParquet(input, dir).coalesce(1)
+      data.createOrReplaceTempView("t")
+      val df = spark.sql(s"select a, xxhash64(a), xxhash64(1, a) from t order by a")
+      checkSparkAnswerAndOperator(df)
+    }
+  }
```

Comment:
Hmm, this is randomly generated, how do we make sure it definitely produces the data causing the error? Could we have a deterministic test case for the fix?

Reply:
We are using a fixed seed for the data generation, and a single partition, so this is deterministic

Reply:
It could be good to also add a specific test with hard-coded values for the problematic use cases though